lib/ukallocbbuddy/bbuddy.c

```diff
@@ -281,6 +331,31 @@ static void *bbuddy_palloc(struct uk_alloc *a, unsigned long num_pages)
 	map_alloc(b, (uintptr_t)alloc_ch, 1UL << order);
 
 	uk_alloc_stats_count_palloc(a, (void *) alloc_ch, num_pages);
+	/* Remove the chunk from the balloon - not for KVM */
+#ifndef CONFIG_PLAT_KVM
```

Comment:
This whole KVM-differenciation thing is really shaky, we need to come up with a better design here, or at least much much better explanations in the code on why this is working like that.